src/RabbitMQ/AmqpAsyncHandler.php

```diff
@@ -136,7 +136,7 @@ public function daemonize()
             $this->logger->info('Starting AMqpAsyncHandler daemon...');
         }
         
-        while (count($this->channel->callbacks)) {
+        while (true) {
```

Comment:
I itend using the callbacks removal as a smart way to end the loop...

So, I really don't like this PR :)